cmd/cli/root.go

```diff
@@ -58,16 +59,16 @@ func NewRootCmd() *cobra.Command {
 				// didn't provide on.
 				util.Fatal(cmd, fmt.Errorf("bacalhau repo not set, please use BACALHAU_DIR or --repo"), 1)
 			}
-			if _, err := setup.SetupBacalhauRepo(repoDir); err != nil {
+			if fsRepo, err := setup.SetupBacalhauRepo(repoDir); err != nil {
 				util.Fatal(cmd, fmt.Errorf("failed to initialize bacalhau repo at '%s': %w", repoDir, err), 1)
+			} else {
+				ctx = context.WithValue(ctx, util.FSRepoKey, fsRepo)
```

Comment:
We need an alternative to embedding things in the context. This pattern isn't sustainable, and documentation states it should be avoided. https://pkg.go.dev/context